### ISSUE TITLE
VTU-next renders additional markup

### ISSUE THREAD

**Issue by afontcu** [vuejs/vue-test-utils-next#219]
Hi folks! 👋 

When trying to port Vue Testing Lib to Vue 3 (and thus vtu-next), I've encountered several issues regarding additional markup rendered by vtu-next.

Some context: Vue Testing Lib provides `container` and `baseElement` [keys](https://testing-library.com/docs/vue-testing-library/api#container), to help testing scenarios where a div is not a valid DOM wrapper element (imagine a table with rows and columns). I first attempted to leverage attachTo ([source](https://github.com/testing-library/vue-testing-library/pull/137/files#diff-2cfdd145f61e8456ea199f8af800c145R66)), but couldn't prevent snapshots from breaking. I'm not that concerned about snapshots _per se_, but rather the implications of wrapping every rendered component with an additional div with custom attributes:

![imatge](https://user-images.githubusercontent.com/9197791/95600806-f04b6100-0a52-11eb-9669-493dac118b98.png)

I'm not entirely sure we need to mount a div with id="app" – looks like one of the first iterations we did when started working with vtu-next, but I'd say it could go now: https://github.com/vuejs/vue-test-utils-next/blob/master/src/mount.ts#L196-L197

Could we remove the additional markup? This markup will likely impact on any snapshot-based testing done with VTU, and also tools that traverse (poorly, I grant you that 😉) the DOM.

Thanks!

**Comment by lmiller1990** [comment]
Hm, Vue needs *something* to mount on, right? Does testing-library use `attachTo` by default?

I agree we probably do not need the `id`.

**Comment by afontcu** [comment]
> Hm, Vue needs _something_ to mount on, right? Does testing-library use `attachTo` by default?

I've tried several combinations of attaching to the container, the baseElement, or a totally unrelated element, without success. Until now (Vue 2), we attach to the `container`, which makes total sense.

Now, if I do something like the following, tests pass:

```js
const wrapper = mount(Component, { attachTo: container }) 
// here, container is the same thing it was in Vue Testing Lib for Vue 2

function unwrap(node) {
  node.replaceWith(...node.childNodes)
}

unwrap(wrapper.parentElement) // this removes the additional "data-v-app" div node
```

but still feels a bit hacky?

**Comment by lmiller1990** [comment]
One difference is that Vue 2 did not have top level fragments; Vue 3 does. I think this is why we have the top level node (maybe).

**Comment by lmiller1990** [comment]
Is this still an issue? Seems you have a fix in testing lib - and #225 removes the useless id.

**Comment by afontcu** [comment]
Not an issue anymoe, I decided to unwrapp the wrapper element ([source](https://github.com/testing-library/vue-testing-library/pull/137/files#diff-4f8b5837e5c68e71852f1c3f41f2e98201426d45a94bd7767cddfc98ccd02fc5R64)) so the snapshots remain as they were :) Closing this up!
